src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -167,6 +168,7 @@ static String guessModByFileName(String fileName) {
 
     try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(replaysDirectory, replayFileGlob)) {
       StreamSupport.stream(directoryStream.spliterator(), false)
+          .sorted(Comparator.comparing(path -> ((Path) path).toFile().lastModified()).reversed())
```

Comment:
Whenever you need a method that only exists in `File` (which is old and bad API), an equivalent method exists in `File` (which is the new, good API).
```suggestion
          .sorted(Comparator.comparing(path -> noCatch(() -> getLastModifiedTime(path))))
```